create_shards.c

```diff
@@ -91,12 +103,69 @@ master_create_distributed_table(PG_FUNCTION_ARGS)
 		ereport(ERROR, (errmsg("could not find column: %s", partitionColumnName)));
 	}
 
-	/* we only support hash partitioning method for now */
-	if (partitionMethod != HASH_PARTITION_TYPE)
+	partitionColumnVar = ColumnNameToColumn(distributedTableId, partitionColumnName);
```

Comment:
Move the call to `ColumnNameToColumn` directly into the declaration of the variable, at the top of this function. Similarly, move the definition of `partitionColumnTypeId` up there.